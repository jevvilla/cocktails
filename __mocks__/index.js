export const drink = {
  details: [
    {
      dateModified: '2016-08-31 19:48:22',
      idDrink: '16108',
      strAlcoholic: 'Alcoholic',
      strCategory: 'Cocktail',
      strCreativeCommonsConfirmed: 'No',
      strDrink: '9 1/2 Weeks',
      strDrinkAlternate: null,
      strDrinkDE: null,
      strDrinkES: null,
      strDrinkFR: null,
      strDrinkThumb: 'https://www.thecocktaildb.com/images/media/drink/xvwusr1472669302.jpg',
      'strDrinkZH-HANS': null,
      'strDrinkZH-HANT': null,
      strGlass: 'Cocktail glass',
      strIBA: null,
      strIngredient1: 'Absolut Citron',
      strIngredient2: 'Orange Curacao',
      strIngredient3: 'Strawberry liqueur',
      strIngredient4: 'Orange juice',
      strIngredient5: null,
      strIngredient6: null,
      strIngredient7: null,
      strIngredient8: null,
      strIngredient9: null,
      strIngredient10: null,
      strIngredient11: null,
      strIngredient12: null,
      strIngredient13: null,
      strIngredient14: null,
      strIngredient15: null,
      strInstructions:
        'Combine all ingredients in glass mixer. Chill and strain into Cocktail glass. Garnish with sliced strawberry.',
      strInstructionsDE:
        'Alle Zutaten in einem Glasmischer vermengen. Abk√ºhlen lassen und in das Cocktailglas abseihen. Mit Erdbeerscheiben garnieren.',
      strInstructionsES: null,
      strInstructionsFR: null,
      'strInstructionsZH-HANS': null,
      'strInstructionsZH-HANT': null,
      strMeasure1: '2 oz ',
      strMeasure2: '1/2 oz ',
      strMeasure3: '1 splash ',
      strMeasure4: '1 oz ',
      strMeasure5: null,
      strMeasure6: null,
      strMeasure7: null,
      strMeasure8: null,
      strMeasure9: null,
      strMeasure10: null,
      strMeasure11: null,
      strMeasure12: null,
      strMeasure13: null,
      strMeasure14: null,
      strMeasure15: null,
      strTags: null,
      strVideo: null,
    },
  ],
  idDrink: '16108',
  strDrink: '9 1/2 Weeks',
  strDrinkThumb: 'https://www.thecocktaildb.com/images/media/drink/xvwusr1472669302.jpg',
};

export const filteredIngredients = [
  'Absolut Citron',
  'Orange Curacao',
  'Strawberry liqueur',
  'Orange juice',
];

export const filteredMeasures = ['2 oz ', '1/2 oz ', '1 splash ', '1 oz '];

export const measuresAndIngredients = [
  '2 oz - Absolut Citron',
  '1/2 oz - Orange Curacao',
  '1 splash - Strawberry liqueur',
  '1 oz - Orange juice',
];
