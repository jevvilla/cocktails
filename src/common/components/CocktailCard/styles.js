import { StyleSheet } from 'react-native';

import colors from '../../colors';
import styles from '../../styles';

export default StyleSheet.create({
  container: {
    ...styles.borderRadius,
    ...styles.padding,
    backgroundColor: colors.white,
    flexDirection: 'row',
    height: 180,
    justifyContent: 'space-between',
  },
  descriptionWrapper: {
    ...styles.paddingRight,
    flex: 1,
  },
  image: {
    ...styles.borderRadius,
    width: 150,
  },
  touchable: {
    ...styles.borderRadius,
    ...styles.marginBottom,
    ...styles.shadow(4),
  },
  title: {
    ...styles.marginBottom,
    color: colors.black,
    fontSize: 22,
    opacity: 0.6,
  },
  ingredient: {
    color: colors.black,
    fontSize: 16,
    opacity: 0.4,
  },
});
