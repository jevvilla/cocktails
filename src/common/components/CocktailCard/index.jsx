import React from 'react';
import PropTypes from 'prop-types';
import { Image, Text, TouchableOpacity, View } from 'react-native';

import styles from './styles';

const renderIngredients = ingredients => {
  return ingredients.map((ingredient, index) => {
    if (index === 2) {
      return <Text style={styles.leftIngredients}>{ingredient}</Text>;
    }
    return <Text style={styles.ingredient}>{`\u2022 ${ingredient}`}</Text>;
  });
};

const CocktailCard = ({ uri, title, onPress, ingredients }) => {
  return (
    <TouchableOpacity onPress={onPress} activeOpacity={0.7} style={styles.touchable}>
      <View style={styles.container}>
        <View style={styles.descriptionWrapper}>
          <Text numberOfLines={2} style={styles.title}>
            {title}
          </Text>
          {renderIngredients(ingredients)}
        </View>
        <Image style={styles.image} source={{ uri }} />
      </View>
    </TouchableOpacity>
  );
};

CocktailCard.propTypes = {
  title: PropTypes.string.isRequired,
  uri: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
  ingredients: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default CocktailCard;
