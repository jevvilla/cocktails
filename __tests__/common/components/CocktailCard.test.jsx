import React from 'react';
import renderer from 'react-test-renderer';

import CocktailCard from '../../../src/common/components/CocktailCard';

import { filteredMeasures } from '../../../__mocks__';

const props = {
  uri: 'https://www.thecocktaildb.com/images/media/drink/xvwusr1472669302.jpg',
  title: '9 1/2 Weeks',
  onPress: jest.fn(),
  ingredients: filteredMeasures,
};

describe('CocktailCard', () => {
  const { uri, title, onPress, ingredients } = props;

  const tree = renderer.create(
    <CocktailCard uri={uri} title={title} onPress={onPress} ingredients={ingredients} />,
  );

  test('should render the time correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
