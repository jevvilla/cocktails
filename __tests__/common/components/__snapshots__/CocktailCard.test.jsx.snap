// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CocktailCard should render the time correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 8,
      "elevation": 4,
      "marginBottom": 16,
      "opacity": 1,
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 4,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderRadius": 8,
        "flexDirection": "row",
        "height": 180,
        "justifyContent": "space-between",
        "padding": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        numberOfLines={2}
        style={
          Object {
            "color": "#000",
            "fontSize": 22,
            "marginBottom": 16,
            "opacity": 0.6,
          }
        }
      >
        9 1/2 Weeks
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "opacity": 0.4,
          }
        }
      >
        • 2 oz 
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "opacity": 0.4,
          }
        }
      >
        • 1/2 oz 
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 12,
            "marginTop": 4,
            "opacity": 0.4,
          }
        }
      >
        1 splash 
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "opacity": 0.4,
          }
        }
      >
        • 1 oz 
      </Text>
    </View>
    <Image
      source={
        Object {
          "uri": "https://www.thecocktaildb.com/images/media/drink/xvwusr1472669302.jpg",
        }
      }
      style={
        Object {
          "borderRadius": 8,
          "width": 150,
        }
      }
    />
  </View>
</View>
`;
